### PR TITLE
Port latest SDK options - React

### PR DESCRIPTION
**Pull request by cristobal-buenrostro**
update sample with latest sdk options

- disableGrammar
- disableTranslation
- disableLanguageDetection
- parent

### CODE REVIEW

File: js/samples/react-sample-simple/src/AppOptions.js [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+        },
+        body: formBodyStr,
+      });
+      const json = await response.json();
+      const { access_token } = json;
+      setToken(access_token)
+
+    }
+    catch (err) {
+      alert('There was a problem fetching your credentials, please check the console and make sure your environment variables are prefixed with REACT_APP_');
```
Comment (rwallerms) [inline js/samples/react-sample-simple/src/AppOptions.js:69]: > make sure your environment variables are prefixed with REACT_APP_

should be explicit checks for expected values up front with specific error message

File: js/samples/react-sample-simple/src/AppOptions.js [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+    // Learn more about options https://docs.microsoft.com/azure/cognitive-services/immersive-reader/reference#options
+    let options = {};
+
+    options.disableLanguageDetection = sampleId === 'DisableLanguageDetection';
+    options.disableGrammar = sampleId === 'DisableGrammar';
+    options.disableTranslation = sampleId === 'DisableTranslation';
+    options.parent = sampleId === 'Parent' && document.getElementById("checkboxParent").checked ? document.getElementById('parentDiv') : null;
+
+    try {
+      await launchAsync(token, process.env.REACT_APP_SUBDOMAIN, data, options)
```
Comment (rwallerms) [inline js/samples/react-sample-simple/src/AppOptions.js:96]: REACT_APP_SUBDOMAIN

check for this up front with explicit error msg

File: js/samples/react-sample-simple/src/AppOptions.js [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+    }
+    catch (error) {
+      alert("Error in launching the Immersive Reader. Check the console.");
+      console.log(error);
+    }
+  }
+
+  const handleLanguage = (e) => setLanguage(e.target.value);
+
+  useEffect(() => {
```
Comment (rwallerms) [inline js/samples/react-sample-simple/src/AppOptions.js:106]: add comment on how useEffect() works

**Review by rwallerms — CHANGES_REQUESTED** [review]

File: js/samples/react-sample-simple/src/AppOptions.js (hunk 3 above)
Comment (cristobal-buenrostro) [reply js/samples/react-sample-simple/src/AppOptions.js:106]: added, similar to the react sample server

File: js/samples/react-sample-simple/src/AppOptions.js (hunk 2 above)
Comment (cristobal-buenrostro) [reply js/samples/react-sample-simple/src/AppOptions.js:96]: done, checking values at getCredentials in constructor

File: js/samples/react-sample-simple/src/AppOptions.js (hunk 1 above)
Comment (cristobal-buenrostro) [reply js/samples/react-sample-simple/src/AppOptions.js:69]: done, checking values at getCredentials in component constructor

File: js/samples/react-sample-simple/src/AppOptions.js [hunk 4]
```diff
@@ -33,6 +33,20 @@ function AppOptions() {
 
   const getCredentials = async () => {
 
+    // Verify environment variables values
+    if( typeof process.env.REACT_APP_CLIENT_ID === 'undefined' || process.env.REACT_APP_CLIENT_ID === null ){
```
Comment (rwallerms) [inline js/samples/react-sample-simple/src/AppOptions.js:37]: FALSY?

**Review by rwallerms — CHANGES_REQUESTED** [review]

File: js/samples/react-sample-simple/src/AppOptions.js (hunk 4 above)
Comment (cristobal-buenrostro) [reply js/samples/react-sample-simple/src/AppOptions.js:37]: done

File: js/samples/react-sample-simple/src/App.js [hunk 5]
```diff
@@ -15,6 +15,20 @@ function App() {
 
   const getCredentials = async () => {
 
+    // Verify environment variables values
+    if( typeof process.env.REACT_APP_CLIENT_ID === 'undefined' || process.env.REACT_APP_CLIENT_ID === null ){
```
[the remark this answers is not in the document]
Comment (cristobal-buenrostro) [reply js/samples/react-sample-simple/src/App.js]: correct, falsy worked

image

**Review by rwallerms — APPROVED** [review]

**Pull request merged**